docs/source/accelerators/gpu_intermediate.rst

```diff
@@ -418,7 +424,7 @@ To illustrate why this is needed, let's look at DataParallel
         x, y = batch
         y_hat = self(batch)
 
-        # on dp or ddp2 if we did softmax now it would be wrong
+        # on dp if we did softmax now it would be wrong
```

Comment:
I know, you just adapted it, but I don't get this, tbh. Shouldn't softmax be per sample of the batch? And since DP splits across batchdim, the samples should all be complete?
Maybe I am just being dumb right now...

Reply:
@williamFalcon wrote this. I can imagine that for things like contrastive learning where both positive and negative samples are included in the batch, you would take multiple of them and combine them in the computational graph together. I do not know if that's what this code is supposed to show.

Reply:
Could be, but I guess, for 99% of the use-cases this is actually not relevant which is why I would remove this to avoid confusion here.